### PR TITLE
N8n 4168 add node ids

### PR DESCRIPTION
**Pull request by mutdmour**

### CODE REVIEW

File: packages/editor-ui/src/components/Node.vue [hunk 1]
```diff
@@ -75,7 +75,7 @@
 		<div class="node-description">
 			<div class="node-name" :title="nodeTitle">
 				<p>
-					{{ nodeTitle }}
+					{{ node.id }}
```
Comment (mutdmour) [inline packages/editor-ui/src/components/Node.vue:78]: TODO REMOVE BEFORE MERGE

File: packages/cli/src/PublicApi/v1/handlers/workflows/workflows.service.ts [hunk 2]
```diff
@@ -133,6 +134,7 @@ export function hasStartNode(workflow: WorkflowEntity): boolean {
 
 export function getStartNode(): INode {
 	return {
+		id: v4(),
```
Comment (BHesseldieck) [inline packages/cli/src/PublicApi/v1/handlers/workflows/workflows.service.ts:137]: why not `uuid()` and importing `v4` twice?

File: packages/cli/src/WorkflowHelpers.ts [hunk 3]
```diff
@@ -474,6 +475,22 @@ export async function getStaticDataById(workflowId: string | number) {
...
+ */
+export function addNodeIds(workflow: WorkflowEntity) {
+	const { nodes } = workflow;
+	if (!nodes) return;
+
+	for (const node of nodes) {
+		if (!node.id) {
+			node.id = uuid();
+		}
+	}
```
Comment (BHesseldieck) [inline packages/cli/src/WorkflowHelpers.ts:491]: nit:
```suggestion
	nodes.forEach((node) => {
		if (!node.id) {
			node.id = uuid();
		}
	});
```

File: packages/cli/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+		// @ts-ignore
+		await runChunked(queryRunner, workflowsQuery, (workflows) => {
+			workflows.forEach(async (workflow) => {
+				const nodes = workflow.nodes;
+				// @ts-ignore
+				nodes.forEach((node) => {
+					if (!node.id) {
+						delete node.id;
+					}
```
Comment (BHesseldieck) [inline packages/cli/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.ts:62]: wrong logic + `delete` is indempotent
```suggestion
				  delete node.id;
```

File: packages/cli/src/databases/migrations/postgresdb/1658932090381-AddNodeIds.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+		// @ts-ignore
+		await runChunked(queryRunner, workflowsQuery, (workflows) => {
+			workflows.forEach(async (workflow) => {
+				const nodes = workflow.nodes;
+				// @ts-ignore
+				nodes.forEach((node) => {
+					if (!node.id) {
+						delete node.id;
+					}
+				});
```
Comment (BHesseldieck) [inline packages/cli/src/databases/migrations/postgresdb/1658932090381-AddNodeIds.ts:75]: wrong logic + `delete` is indempotent 👍🏼 
```suggestion
				nodes.forEach((node) => delete node.id);
```

File: packages/cli/src/databases/migrations/sqlite/1658930531669-AddNodeIds.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+		`;
+
+		// @ts-ignore
+		await runChunked(queryRunner, workflowsQuery, (workflows) => {
+			workflows.forEach(async (workflow) => {
+				const nodes = JSON.parse(workflow.nodes);
+				nodes.forEach((node: INode) => {
+					// @ts-ignore
+					delete node.id;
+				});
```
Comment (BHesseldieck) [inline packages/cli/src/databases/migrations/sqlite/1658930531669-AddNodeIds.ts:67]: nit:
```suggestion
				nodes.forEach((node: INode) => delete node.id);
```

File: packages/editor-ui/src/views/canvasHelpers.ts [hunk 7]
```diff
@@ -10,6 +10,7 @@ import {
 	NodeInputConnections,
 	INodeTypeDescription,
 } from 'n8n-workflow';
+import { v4 as uuidv4} from 'uuid';
```
Comment (BHesseldieck) [inline packages/editor-ui/src/views/canvasHelpers.ts:13]: nit:
```suggestion
import { v4 as uuid } from 'uuid';
```

File: packages/editor-ui/src/views/canvasHelpers.ts [hunk 8]
```diff
@@ -728,7 +729,7 @@ export const getFixedNodesList = (workflowNodes: INode[]) => {
 	});
 
 	if (!hasStartNode) {
-		nodes.push({...DEFAULT_START_NODE});
+		nodes.push({...DEFAULT_START_NODE, id: uuidv4() });
```
Comment (BHesseldieck) [inline packages/editor-ui/src/views/canvasHelpers.ts:732]: ref previous comment
```suggestion
		nodes.push({...DEFAULT_START_NODE, id: uuid() });
```

**Review by BHesseldieck — CHANGES_REQUESTED** [review]

**Review by BHesseldieck — APPROVED** [review]

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with `n8n@0.189.0`

**Pull request merged**